akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -239,6 +240,21 @@ object HttpEntity {
       else Default(contentType, data.length, limitableByteSource(Source.single(data))) withSizeLimit maxBytes
 
     override def productPrefix = "HttpEntity.Strict"
+
+    override def toString = {
+      val dataAsString = contentType match {
+        case _: Binary ⇒
+          data.toString()
+        case nb: NonBinary ⇒
+          val decodedString = data.decodeString(nb.charset.value)
+          if (decodedString.length > 4000)
```

Comment:
Should it maybe not decode the entire body first but limit the bytes it will make a string out of, imagine a 10mb body, copied into a string just to take the first 4000 of it.